### ISSUE TITLE
Cannot load package because nuget creates two nuspec files

### ISSUE THREAD

**Issue by Emik03** [waf/CSharpRepl#259]
### Version

0.6.1

## Problem

Trying to reference a package that references another can cause the error shown below.

You may need to run `csharprepl -r "nuget: Emik.Unions, 2.0.1"` twice for it to break, sometimes the first pass works.

```
emik@spicy-linux:~/RiderProjects/Emik.Morsels/Experimental$ csharprepl -r "nuget: Emik.Unions, 2.0.1"
Welcome to the C# REPL (Read Eval Print Loop)!
Type C# expressions and statements at the prompt and press Enter to evaluate them.
Type help to learn more, and type exit to quit.

Adding supplied references...
Could not find package 'Emik.Unions'. Error: NuGet.Packaging.Core.PackagingException: Package contains multiple nuspec files.
   at NuGet.Packaging.PackageFolderReader.GetNuspecFile()
   at NuGet.Packaging.PackageReaderBase.GetNuspec()
   at NuGet.Packaging.PackageReaderBase.get_NuspecReader()
   at NuGet.Packaging.PackageReaderBase.GetIdentity()
   at NuGet.Packaging.PackageReaderBase.GetBuildItems()
   at NuGet.Packaging.PackageReaderBase.GetSupportedFrameworks()
   at NuGet.Packaging.PackageReaderBase.GetSupportedFrameworksAsync(CancellationToken cancellationToken)
   at CSharpRepl.Services.Nuget.NugetPackageInstaller.GetDependencies(NuGetFramework targetFramework, FolderNuGetProject nuGetProject, PackageIdentity 
packageIdentity, ManagedCodeConventions managedCodeConventions, Dictionary`2 aggregatedReferences, CancellationToken cancellationToken) in 
D:\a\CSharpRepl\CSharpRepl\CSharpRepl.Services\Nuget\NugetPackageInstaller.cs:line 165
   at CSharpRepl.Services.Nuget.NugetPackageInstaller.GetDependencies(NuGetFramework targetFramework, FolderNuGetProject nuGetProject, PackageIdentity 
packageIdentity, ManagedCodeConventions managedCodeConventions, Dictionary`2 aggregatedReferences, CancellationToken cancellationToken) in 
D:\a\CSharpRepl\CSharpRepl\CSharpRepl.Services\Nuget\NugetPackageInstaller.cs:line 205
   at CSharpRepl.Services.Nuget.NugetPackageInstaller.GetDependencies(NuGetFramework targetFramework, FolderNuGetProject nuGetProject, PackageIdentity 
packageIdentity, ManagedCodeConventions managedCodeConventions, CancellationToken cancellationToken) in 
D:\a\CSharpRepl\CSharpRepl\CSharpRepl.Services\Nuget\NugetPackageInstaller.cs:line 127
   at CSharpRepl.Services.Nuget.NugetPackageInstaller.GetAssemblyReferenceWithDependencies(NuGetFramework targetFramework, FolderNuGetProject nuGetProject, 
PackageIdentity packageIdentity, CancellationToken cancellationToken) in D:\a\CSharpRepl\CSharpRepl\CSharpRepl.Services\Nuget\NugetPackageInstaller.cs:line 
115
   at CSharpRepl.Services.Nuget.NugetPackageInstaller.InstallAsync(String packageId, String version, CancellationToken cancellationToken) in 
D:\a\CSharpRepl\CSharpRepl\CSharpRepl.Services\Nuget\NugetPackageInstaller.cs:line 88

> 
```

## Cause

[`Emik.Unions`](https://www.nuget.org/packages/Emik.Unions) references [`Emik.Results`](https://www.nuget.org/packages/Emik.Results), causing the same file with different casing to show up twice, hence the exception `Package contains multiple nuspec files`. It appears to only happen on libraries that are referenced in others, when those are consumed. This error doesn't happen on case-insensitive filesystems.

![image](https://github.com/waf/CSharpRepl/assets/14614115/de6e9fb6-f030-4f08-bc4d-00aa99e414f5)

## Workaround

Deleting either of `Emik.Results.nuspec` or `emik.results.nuspec` fixes the issue.

**Comment by waf** [comment]
Thanks. CSharpRepl doesn't actually manage this folder, and relies on the [Nuget Client Tools](https://github.com/NuGet/NuGet.Client/) nuget packages. We found a very similar issue before (https://github.com/waf/CSharpRepl/issues/52) where we added a workaround. I've set this workaround to handle this specific case too.

**Comment by waf** [comment]
I believe this should be fixed now in v0.6.3 (you can update via `dotnet tool update -g csharprepl`). I wasn't able to reproduce this even on Linux with a case-sensitive filesystem and trying multiple times, so the fix is a bit of a shot-in-the-dark. Please give it a go! Thanks.

**Comment by Emik03** [comment]
I can confirm that the latest version fixes this, huge thank you!
